presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -72,12 +76,15 @@ private static SymbolStatsEstimate addColumnStats(SymbolStatsEstimate leftStats,
         double nullsCountLeft = leftStats.getNullsFraction() * leftRows;
         double totalSizeLeft = (leftRows - nullsCountLeft) * leftStats.getAverageRowSize();
         double totalSizeRight = (rightRows - nullsCountRight) * rightStats.getAverageRowSize();
-        double newNullsFraction = (nullsCountLeft + nullsCountRight) / newRowCount;
+        double newNullsFraction = newRowCount == 0 ? 1.0 : ((nullsCountLeft + nullsCountRight) / newRowCount);
```

Comment:
Maybe we need `firstFinite(double... values)` method that would filter out `-Infinity, +Infinity` values

Reply:
I think being more explicit is better. It is easier to reason about the code then.